lib/buildingsync/get_bcl_weather_file.rb

```diff
@@ -0,0 +1,203 @@
+          if attribute.name == 'WMO'
+            wmo_no = attribute.valueAsDouble
+          end
+        end
+      end
+
+      if choices.count == 0
+        p "Error, could not find uid for state #{state} and city #{city}. Initial count of weather files: #{responses.count}. Please try a different weather file."
+        return false
+      end
+
+      epw_path = download_weather_file(remote, choices)
+      download_design_day_file(wmo_no, epw_path)
+      return epw_path
+  end
+
+    def download_weather_file_from_weather_id(weather_id)
+      wmo_no = 0
+      remote = OpenStudio::RemoteBCL.new
+
+      # Search for weather files
+      responses = remote.searchComponentLibrary(weather_id, 'Weather File')
+
+      # Create options for user prompt
+
+      name_to_uid = Hash.new
+
+      choices = OpenStudio::StringVector.new
+
+      responses.each do |response|
+        if response.name.include? 'TMY3'
+          choices << response.uid
+          name_to_uid[response.name] = response.uid
+
+          response.attributes.each do |attribute|
+            if attribute.name == 'WMO'
+              wmo_no = attribute.valueAsDouble
+            end
+          end
+        end
+      end
+
+      if choices.count == 0
+        p "Error, could not find uid for #{name.valueAsString}.  Please try a different weather file."
+        return false
+      end
+
+      epw_path = download_weather_file(remote, choices)
+      download_design_day_file(wmo_no, epw_path)
+      return epw_path
+    end
+
+    def download_weather_file(remote, choices)
+      epw_path = ''
+
+      choices.each do |choice|
+        uid = choice
+
+        remote.downloadComponent(uid)
+        component = remote.waitForComponentDownload
+
+        if component.empty?
+          p 'Cannot find local component'
+          # runner.registerError('Cannot find local component')
+          return false
+        end
+
+        component = component.get
+
+        files = component.files('epw')
+
+        if files.empty?
+          p 'No epw file found'
+          return false
+        end
+
+        epw_path = component.files('epw')[0]
+      end
+
+      p "Successfully set weather file to #{epw_path}"
+
+      return epw_path
+    end
+
+    def download_design_day_file(wmo_no, epw_path)
+      remote = OpenStudio::RemoteBCL.new
+      responses = remote.searchComponentLibrary(wmo_no.to_s[0, 6], 'Design Day')
+      choices = OpenStudio::StringVector.new
+
+      idf_path_collection = []
+
+      responses.each do |response|
+        choices << response.uid
+      end
+
+      choices.each do |choice|
+        uid = choice
+
+        remote.downloadComponent(uid)
+        component = remote.waitForComponentDownload
+
+        if !component.empty?
+
+          component = component.get
+
+          files = component.files('idf')
+
+          if !files.empty?
+            idf_path_collection.push(component.files('idf')[0])
+          else
+            p 'No idf file found'
```

Comment:
This is nice if we are using this gem as a command line tool, but in the context of a gem, we should have this fail if it is not found and/or log out the error. I suggest changing this to `logger.warn...`, `OpenStudio.logFree(...)`, or `raise Exception...`

Reply:
done